src/cli/api/local_api.py

```diff
@@ -382,20 +526,266 @@ def get_results(self, path: Path, all_results: bool) -> List[ResultType]:
+
+        # Check if the channels exist and belong to this network
+        invalid_channels = []
+        for channel in experiment_channels:
+            # Replace '_' with '-'. Eg: the_hague can be written this way in experiment.json while the comparison is
+            # done with the-hague defined in network/channels.json
+            node_slug1 = channel['node_slug1'].replace('_', '-')
+            node_slug2 = channel['node_slug2'].replace('_', '-')
+            if not node_slug1 + '-' + node_slug2 in network_channels:
+                if not node_slug2 + '-' + node_slug1 in network_channels:
+                    invalid_channels.append([channel['node_slug1'], channel['node_slug2']])
+
+        if invalid_channels:
+            if len(invalid_channels) > 1:
+                error_dict['warning'].append(f"In file {experiment_file_path}: {invalid_channels} do not exist or are "
+                                             f"not valid channels for network '{experiment_network_slug}'")
+            elif len(invalid_channels) == 1:
+                error_dict['warning'].append(f"In file {experiment_file_path}: {invalid_channels} does not exist or is "
+                                             f"not a valid channel for network '{experiment_network_slug}'")
+
+    def validate_experiment_application(self, path: Path, experiment_data: Dict[str, Any],
+                                        error_dict: ErrorDictType) -> None:
+        """
+        Validate the ['application'] key defined in the asset of experiment.json. Check if the roles double the roles
+        that are defined in input/network.json
+
+        Args:
+            path: The location of the experiment
+            experiment_data: contents of the experiment.json file
+            error_dict: Dictionary containing error and warning messages of the validations that failed
+        """
+
+        experiment_application_data = experiment_data['asset']['application']
+        experiment_input_path = path / 'input'
+        network_schema_path = Path(os.path.join(BASE_DIR, 'schema', 'applications', 'network.json', ''))
+
+        # Check if the roles defined in experiment.json ['application'] double the roles defined in input/network.json
+        if (experiment_input_path / 'network.json').is_file():
+            is_network_json_valid, _ = validate_json_file(experiment_input_path / 'network.json')
+            if is_network_json_valid:
+                is_network_schema_valid, _ = validate_json_schema(experiment_input_path / 'network.json',
+                                                                  network_schema_path)
+                if is_network_schema_valid:
+                    application_network_data = read_json_file(experiment_input_path / 'network.json')
+                    for item in experiment_application_data:
+                        experiment_roles = item["roles"]
+                        application_roles = application_network_data['roles']
```

Comment:
These roles are fixed and can be get once. This line can move up to above: for item in experiment_application_data: